### PR TITLE
chore: create propose new rule form

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).
- [ ] If some rule is added/updated/removed, I've regenerated the rules list.
- [ ] If some rule meta info is changed, I've regenerated the plugin shared configs.

## Changes

- Create the form that contributors can use to propose a new rule

## Context

Go to https://github.com/HonkingGoose/eslint-plugin-testing-library/blob/chore/430-propose-new-rule-form/.github/ISSUE_TEMPLATE/propose_new_rule.yml to see a preview of the template.

You can click one of the checkboxes in issue #430 after you've merged this.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      placeholder: The rule should ...
+    validations:
+      required: true
+
+  - type: textarea
+    id: testing_library_feature
+    attributes:
+      label: Testing library feature
+      description: What Testing Library feature does this rule relate to?
+      placeholder: ...
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:28]: I'm not sure what the placeholder text should be here.

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    id: testing_library_frameworks
+    attributes:
+      label: Testing library framework(s)
+      description: What Testing Library framework(s) does this rule relate to?
+      placeholder: ...
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:37]: I'm also not sure what the placeholder text should be here.

**Review by HonkingGoose — COMMENTED** [review]
Ready for first round of review.

I need some help with the `placeholder` text.

Feel free to suggest changes to the text of any field. 😉

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+name: Propose a new rule
+description: Propose a new rule for the eslint-plugin-testing-library.
+labels: [new rule]
+body:
+  - type: input
+    id: name_for_new_rule
+    attributes:
+      label: Name for new rule
+      description: Suggest a name for the new rule.
```
Comment (Belco90) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:9]: Can we add links here? Would be amazing if we could attach the [Rule naming conventions from CONTRIBUTING](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md#rule-naming-conventions)

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      placeholder: Put your suggestion for the name in here.
```
Comment (Belco90) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:10]:
```suggestion
      placeholder: prefer-wait-for
```

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      label: Description of the new rule
+      description: Please describe what the new rule should do.
+      placeholder: The rule should ...
+    validations:
+      required: true
+
+  - type: textarea
+    id: testing_library_feature
+    attributes:
+      label: Testing library feature
```
Comment (Belco90) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:26]:
```suggestion
      label: Testing Library feature
```

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      description: What Testing Library feature does this rule relate to?
+      placeholder: ...
+    validations:
+      required: true
+
+  - type: textarea
+    id: testing_library_frameworks
+    attributes:
+      label: Testing library framework(s)
```
Comment (Belco90) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:35]:
```suggestion
      label: Testing Library framework(s)
```

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml (hunk 2 above)
Comment (Belco90) [reply .github/ISSUE_TEMPLATE/propose_new_rule.yml:37]:
```suggestion
      placeholder: angular, react and vue
```

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml (hunk 1 above)
Comment (Belco90) [reply .github/ISSUE_TEMPLATE/propose_new_rule.yml:28]: I guess this one could be fine without a placeholder, but we can put an existing rule description:
```suggestion
      placeholder: Enforce promises from async queries to be handled
```

**Review by Belco90 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml (hunk 3 above)
Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/propose_new_rule.yml:9]:
```suggestion
      description: Suggest a name for the new rule that follows the [rule naming conventions](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md#rule-naming-conventions).
```

File: .github/ISSUE_TEMPLATE/propose_new_rule.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      placeholder: Enforce promises from async queries to be handled
+    validations:
+      required: true
+
+  - type: textarea
+    id: testing_library_frameworks
+    attributes:
+      label: Testing Library framework(s)
+      description: What Testing Library framework(s) does this rule relate to?
+      placeholder: angular, react and vue
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/propose_new_rule.yml:37]:
```suggestion
      placeholder: Angular, React and Vue
```

We should probably capitalize these names. 😄

Comment (Belco90) [reply .github/ISSUE_TEMPLATE/propose_new_rule.yml:37]: Sure!

**Review by Belco90 — APPROVED** [review]

**Comment by Belco90** [comment]
Let's merge the forms then!

**Pull request merged**